### ISSUE TITLE
Handle timestamps correctly

### ISSUE THREAD

**Issue by qtrandev** [CodeForFtL/ftlviz#31]
I'm having problem with displaying the timestamps correctly. This may be due to the time data being converted to Eastern Standard Time, so 2018 becomes 5 hours earlier which is 12/31/2017.

For example if you click on the year "2018", the range of time is "Sunday, 31st December 2017 to Monday, 31st December 2018"

Maybe use a library like Moment.js to help. I'm not sure of the direction to go with this. We can assume all the dates given is Fort Lauderdale date. We can also assume all users are from Florida.

![image](https://cloud.githubusercontent.com/assets/1198220/11860481/518b5518-a443-11e5-848f-b89da3e32788.png)

**Comment by lodenrogue** [comment]
Just so I understand the issue: When the dates are displayed starting from 31st December 20XX and ending in any date with the year 20XX + y then you want it to display 31st December 20XX + 1. Is that correct?

If so (pseudo code):

if(startDate.getDay() == 31 && startDate.getMonth() == "December"){
   display(startDate.getDay() + startDate.getMonth() + (startDate.getYear() + 1));
}

(end pseudo code)

So basically just add 1 to the year if it's the 31st of December. Unless you don't have access to the code and you're relying on some library to handle how you display the time stamp.

**Comment by lodenrogue** [comment]
Ok. I think I understand the problem: When the dates are displayed starting from 31st December 20XX and ending in any date with the year 20XX + y then you want it to display 1st January 20XX + 1. Is that correct?

if so then just do a check for start date and end date. If start date day is 31st and start date month is December; then, if end date year is greater than start date year then change start date to new date(1, "January", startDate.getYear() + 1).

**BUT** you only do this when one of the preset ranges is clicked. Why? Because the user may actually want to manually move the slider to display Dec 31st 20XX.

**Comment by qtrandev** [comment]
Thanks for the feedback! Let me look at it later to see what my actual problem is. I need to clarify that. There was some reason why I had to hack it to show "December 31st 2017".
